{"metadata":{"kernelspec":{"display_name":"Bash","language":"bash","name":"bash"},"language_info":{"codemirror_mode":"shell","file_extension":".sh","mimetype":"text/x-sh","name":"bash"}},"nbformat_minor":5,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# Guts notebook","metadata":{"tags":[]},"id":"36d773bc-63cf-439e-b38a-927b8d701177"},{"cell_type":"markdown","source":"![guts](assets/img/guts.png)","metadata":{},"id":"3f408b0d-db7a-4300-8500-fd7581b2c487"},{"cell_type":"markdown","source":"This datasets contains stomach contents from sardines and sprats in the Bay of Biscay.\n\n- In total there are 144 samples (too many).\n\n- There are 5 groups of samples:\n  - break: sardines from the shelf break (A)\n  - outer: sardines from hte outer shelf (B)\n  - day: sardines from the inner shelf taken by the day\n  - night: sardines from the inner shelf taken at night\n  - sprat: sprats from the inner shelf\n\n- But we don't want to study bacteria but the zooplankton they eat. Therefore we don't use the 16S marker but the 18S.\n\n","metadata":{},"id":"7b234e31-16e0-46c3-89a7-6248127575c4"},{"cell_type":"markdown","source":"Your task is to study the differences in the diet between a pair of sample types.","metadata":{"tags":[]},"id":"70af491c-a588-4dbc-985a-50e1206eec80"},{"cell_type":"markdown","source":"This is who has to do what:","metadata":{},"id":"3633812f-b05c-452a-a233-c5feaf70eaf9"},{"cell_type":"markdown","source":"| Student   | Pair of samples | sample_data      |\n|-----------|-----------------|------------------|\n| Hossain   | Break vs night  | guts_break_night |\n| Gema      | Break vs outer  | guts_break_outer |\n| Madeleine | Break vs sprat  | guts_break_sprat |\n| Samuel    | Night vs outer  | guts_night_outer |\n| Borja     | Night vs sprat  | guts_night_sprat |\n| Gaet√°n    | Outer vs sprat  | guts_outer_sprat |","metadata":{},"id":"5a15d8cd-a25f-4e16-982c-63e75ec5618d"},{"cell_type":"markdown","source":"Feel free to have last week's notebooks open in another tab.","metadata":{},"id":"589f2e60-ba43-4633-aa83-81fad04b65b2"},{"cell_type":"markdown","source":"The files you have to analyze are in the nervion folder:","metadata":{},"id":"9015998e-a694-42e5-acd8-697a11da9251"},{"cell_type":"code","source":"ll -h guts","metadata":{"tags":[],"trusted":true},"execution_count":1,"outputs":[{"name":"stdout","text":"total 48M\ndrwxr-xr-x 2 jovyan jovyan 4.0K Mar 31 14:31 ./\ndrwxr-xr-x 1 jovyan jovyan 4.0K Apr 19 18:28 ../\n-rw-r--r-- 1 jovyan jovyan 3.6M Mar 31 14:31 barcodes.fastq.gz\n-rw-r--r-- 1 jovyan jovyan  23M Mar 31 14:31 forward.fastq.gz\n-rw-r--r-- 1 jovyan jovyan  22M Mar 31 14:31 reverse.fastq.gz\n","output_type":"stream"}],"id":"867ae067-b7be-429f-adb2-c68eb636d7ca"},{"cell_type":"markdown","source":"It is a decaffeinated dataset from the one from the article (~ 0.5 to 1%) so you can analyze in a moment, rather than waiting a day or two to then realize that you did it wrong.","metadata":{},"id":"20c3c447-7d16-46d1-ac80-c47cbfee5bea"},{"cell_type":"markdown","source":"Insteado of sequencing in a single go the 16S sequence, we sequence twice the fragment, once in the 5' to 3' direction (`forward.fastq.gz`), and again in the 3' to 5' sense (`reverse.fastq.gz`). By sequencing twice we get more accurate measurements. ","metadata":{},"id":"92633376-f96d-4821-90a0-e37d80c9e5e9"},{"cell_type":"markdown","source":"## Warning","metadata":{},"id":"4ec1d231-02e9-4a60-a038-f4f0c5b5608c"},{"cell_type":"markdown","source":"The only difference with the dataset from the previous session is that we are using Illumina **Paired**-End sequencing instead of **Single**-End.","metadata":{},"id":"6b0600b8-de14-4d01-9c09-c829955a783c"},{"cell_type":"markdown","source":"Therefore, you will need to modify the commands specific to single-end to paired end, i.e.:\n- `qiime demux emp-single` now becomes `qiime demux emp-paired`\n- `qiime dada2 denoise-single` is now `qiime dada2 denoise-paired`","metadata":{"tags":[]},"id":"e8575d6b-31e2-49e6-8c63-16417cdcd170"},{"cell_type":"markdown","source":"Appart from that, almost every command is the same.","metadata":{},"id":"be902ea0-53a0-4aae-ae3f-9b8c7ff6cb7b"},{"cell_type":"markdown","source":"The purpose of this exercise is to see that you can follow from the start to the end  ","metadata":{},"id":"e0379f9b-0484-429b-9d52-521964b0572f"},{"cell_type":"markdown","source":"## Tips\n\n- Save often, and download the notebook to your laptop from time to time.\n\n- Remember to `pause` your notebook if you think you need to rest or to think.\n\n- Click on the Fast Forward button from time to time (between restart and download buttons in the top bar) to see that everything works from beginning to end","metadata":{},"id":"8e6e606b-131b-42c0-b9a6-1d5380e83d96"},{"cell_type":"markdown","source":"You are expected to type commands in code cells like the following one with a `[ ]` on the side","metadata":{"tags":[]},"id":"80529fa8-b15a-4e07-8bc5-6cc6e95b1d06"},{"cell_type":"code","source":"","metadata":{},"execution_count":null,"outputs":[],"id":"d449a7b2-e172-4df7-abf7-9244bb04197b"},{"cell_type":"markdown","source":"and text in the cells without the `[ ]`:","metadata":{},"id":"da274457-3cf6-4ce1-9e74-21e8e14578f5"},{"cell_type":"markdown","source":"","metadata":{},"id":"f76c94bf-55c7-4891-b2ba-812ae519fc6e"},{"cell_type":"markdown","source":"If you find two or more code cells together, is because you are being asked for more than one command. Put each command in different cells","metadata":{},"id":"4af4a551-55be-4681-ab1c-70e965e06267"},{"cell_type":"markdown","source":"# Exercises","metadata":{},"id":"5807b96e-7d56-478b-9b0a-f4ffb2835272"},{"cell_type":"markdown","source":"## 1. Samples","metadata":{},"id":"92a8ee03-4108-4aca-86fc-3e5ce9457de4"},{"cell_type":"markdown","source":"Let's see if you understand the contents of your `sample-data.tsv` file.","metadata":{"tags":[]},"id":"95b33a97-e8dd-4b52-a121-8a7450fe12bb"},{"cell_type":"markdown","source":"1.1 How many samples are in your `sample-data.tsv` file?","metadata":{},"id":"bdc653e9-551b-4966-aee4-af3aba4e1e5f"},{"cell_type":"markdown","source":"There are 43 samples (guts_outer)","metadata":{},"id":"1c7a3e5c-19b1-4b27-80b2-3b77ad1b9475"},{"cell_type":"markdown","source":"1.2 How many are for each city?","metadata":{},"id":"2dcad554-946b-4e02-8658-02c68863e2f4"},{"cell_type":"markdown","source":"15 samples from Gironde, and 27 from Capbreton","metadata":{},"id":"7e1be192-d71a-4b13-a12e-f2783bc8747e"},{"cell_type":"markdown","source":"1.3 How many samples are for each season of the year?","metadata":{},"id":"d6c8520c-f10a-4aa6-b608-56822f4478ba"},{"cell_type":"markdown","source":"15 samples of day and 27 of night","metadata":{},"id":"e0f74c73-5dfa-4cd4-87c1-193a733d0df0"},{"cell_type":"markdown","source":"1.4 What name has the column that holds the barcode of the sample?","metadata":{},"id":"77bacbce-65a1-4501-b096-c3881c97ce59"},{"cell_type":"markdown","source":"BarcodeSequence","metadata":{},"id":"db3e0964-5041-40bd-9f47-5b1977b5fc7b"},{"cell_type":"markdown","source":"## 2. Pre-processing","metadata":{"tags":[]},"id":"7a455149-4811-46d4-8934-e88299789369"},{"cell_type":"markdown","source":"2.1 Import your dataset to Qiime2. It is not of the type `EMPSingleEndSequences`.","metadata":{},"id":"32fb88e0-1a3b-40dc-aaea-35c277e1223d"},{"cell_type":"code","source":"qiime tools import \\\n  --type        EMPPairedEndSequences \\\n  --input-path  guts \\\n  --output-path sequences.qza","metadata":{"tags":[],"trusted":true},"execution_count":2,"outputs":[{"name":"stdout","text":"\u001b[32mImported guts as EMPPairedEndDirFmt to sequences.qza\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"d715f2b2-b01b-40cd-97b9-a3d19fbdd188"},{"cell_type":"markdown","source":"2.2 Demultiplex your sequences with `demux`. Remember that we are not doing Paired-End sequencing. Also, you need to pass the `--p-rev-comp-mapping-barcodes` parameter.","metadata":{},"id":"92a5a231-e84e-42ba-9066-ffd53388e113"},{"cell_type":"code","source":"qiime demux emp-paired \\\n    --i-seqs                     sequences.qza \\\n    --m-barcodes-file            guts_night_outer.tsv \\\n    --m-barcodes-column          BarcodeSequence \\\n    --p-rev-comp-mapping-barcodes \\\n    --o-per-sample-sequences     demux.qza \\\n    --o-error-correction-details demux-details.qza\n   ","metadata":{"tags":[],"trusted":true},"execution_count":3,"outputs":[{"name":"stdout","text":"\u001b[32mSaved SampleData[PairedEndSequencesWithQuality] to: demux.qza\u001b[0m\n\u001b[32mSaved ErrorCorrectionDetails to: demux-details.qza\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"5680ef6b-e03e-4ac1-8eaf-56be3811214b"},{"cell_type":"markdown","source":"2.3 Generate the `demux.qzv`file and visualize it in [view.qiime.org](view.qiime.org)","metadata":{},"id":"18899f1d-f445-4885-99ce-685781946a39"},{"cell_type":"code","source":"qiime demux summarize \\\n    --i-data          demux.qza \\\n    --o-visualization demux.qzv","metadata":{"tags":[],"trusted":true},"execution_count":4,"outputs":[{"name":"stdout","text":"\u001b[32mSaved Visualization to: demux.qzv\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"f662f785-0c50-44ae-bc12-856edb261720"},{"cell_type":"markdown","source":"2.4 How many samples contains more than 1,000 reads?","metadata":{},"id":"e7e3e3ae-d31a-4ed3-9781-2787d8b6f0f8"},{"cell_type":"markdown","source":"There are 38 samples that have more than 1000 reads","metadata":{},"id":"48c37303-cf63-46f7-a402-d06a5779d579"},{"cell_type":"markdown","source":"2.5 What is the name of the smallest sample?","metadata":{},"id":"ff0ab3e7-3b4a-4b65-a0fe-14f00cf97b47"},{"cell_type":"markdown","source":"sardine_gironde_inner_night_L14.1 (178)","metadata":{},"id":"dfd72e8b-0777-4c6f-a6e1-0d9ebc8d03cf"},{"cell_type":"markdown","source":"2.6 And the biggest?","metadata":{},"id":"698210c0-f9dc-4716-bbda-8060d7c0aa84"},{"cell_type":"markdown","source":"sardine_gironde_inner_night_K95.10.1 (4937)","metadata":{},"id":"f6bbc852-afd6-4c01-816a-0ebe5275fabf"},{"cell_type":"markdown","source":"Click the \"Interactive Quality Plot\" tab.","metadata":{},"id":"b5e36f5f-12ea-4f37-a8c7-582506933076"},{"cell_type":"markdown","source":"2.7 What is the median length of the forward reads?","metadata":{},"id":"49ad604f-8f85-4767-8874-954e67320392"},{"cell_type":"markdown","source":"151 nts","metadata":{},"id":"c2e314b6-ce0b-432b-b397-91dbf09adcbc"},{"cell_type":"markdown","source":"2.8 And the mean quality of the forward reads?","metadata":{},"id":"9c2d03e0-3c2f-4884-98d9-3fdd59efc411"},{"cell_type":"markdown","source":"38","metadata":{},"id":"504d0bfb-23ea-4e2a-8199-ad4758a63198"},{"cell_type":"markdown","source":"2.9 What is the median length of the reverse reads?","metadata":{},"id":"7e3a2836-e282-47a4-ba66-9c36b2447545"},{"cell_type":"markdown","source":"151 nts","metadata":{},"id":"b54daed8-421e-4327-aea2-4bdf561f2b0c"},{"cell_type":"markdown","source":"2.10 And the mean quality of the reverse reads?","metadata":{},"id":"2d7d6b91-875a-468c-996f-965fe78150e0"},{"cell_type":"markdown","source":"39","metadata":{},"id":"384040f1-eca6-4235-a268-8eaf956cc40a"},{"cell_type":"markdown","source":"2.11 Do these reads need to be trimmed either at the 5' or 3' end in order to get a high quality experiment? High quality means a score above 30 (probability of error < 0.001),","metadata":{},"id":"705aaad0-af7d-4a5b-93d4-6124247e141f"},{"cell_type":"markdown","source":"There is no need since both reads have a quality larger than 30","metadata":{},"id":"9ca283bc-9bcb-4490-88aa-78cdcdb5c5ee"},{"cell_type":"markdown","source":"2.12 Perform error correction with DADA2","metadata":{},"id":"f12c2e34-cd81-4d7e-9414-fd38b56c70c5"},{"cell_type":"code","source":"qiime dada2 denoise-paired \\\n    --i-demultiplexed-seqs       demux.qza \\\n    --p-trunc-len-f                151 \\\n    --p-trunc-len-r                151 \\\n    --o-representative-sequences rep-seqs.qza \\\n    --o-table                    table.qza   \\\n    --o-denoising-stats          dada2-stats.qza\n     ","metadata":{"tags":[],"trusted":true},"execution_count":5,"outputs":[{"name":"stdout","text":"\u001b[32mSaved FeatureTable[Frequency] to: table.qza\u001b[0m\n\u001b[32mSaved FeatureData[Sequence] to: rep-seqs.qza\u001b[0m\n\u001b[32mSaved SampleData[DADA2Stats] to: dada2-stats.qza\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"24bb3b1c-0d21-4801-96f2-36036474b155"},{"cell_type":"markdown","source":"2.13 Convert the feature table (`table.qza`) to qzv and open it in the visualizer.","metadata":{},"id":"650775bf-514f-41bb-87ce-fe842b04db4e"},{"cell_type":"code","source":"qiime feature-table summarize \\\n    --i-table                table.qza \\\n    --m-sample-metadata-file guts_night_outer.tsv \\\n    --o-visualization        table.qzv","metadata":{"tags":[],"trusted":true},"execution_count":6,"outputs":[{"name":"stdout","text":"\u001b[32mSaved Visualization to: table.qzv\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"a77db664-d5fb-4426-a983-25f044727b97"},{"cell_type":"markdown","source":"2.14 How many ASVs have you found?","metadata":{"tags":[]},"id":"f172a75e-4d0d-4c4a-8a5f-4d2babf60248"},{"cell_type":"markdown","source":"111","metadata":{},"id":"c77241ea-5299-4438-92d8-6e7c90c859ec"},{"cell_type":"markdown","source":"2.15 How many reads are in total after error correction?","metadata":{},"id":"56623498-4b10-4735-be75-fc4ea9c0638e"},{"cell_type":"markdown","source":"63139","metadata":{},"id":"e181d333-821c-4f07-b90b-934de9ea00c8"},{"cell_type":"markdown","source":"2.16 How many reads has the biggest sample?","metadata":{},"id":"f97efd18-5d65-425b-b37a-3102096613f0"},{"cell_type":"markdown","source":"3202, sardine_capbreton_outer_day_L36.10.1","metadata":{},"id":"a90fcadd-579c-4d01-9224-030c6bf63ee1"},{"cell_type":"markdown","source":"2.17 How many reads has the smallest sample?","metadata":{},"id":"53058b92-e514-49fc-8210-2b06d3924a19"},{"cell_type":"markdown","source":"5, sardine_capbreton_outer_day_L62","metadata":{},"id":"c3981133-c47c-4d37-a7c9-8f24e4d288f0"},{"cell_type":"markdown","source":"2.18 How many samples have more than 1,000 reads?","metadata":{},"id":"726e4ba9-9217-4811-bb7c-b8e6a3ee927a"},{"cell_type":"markdown","source":"30","metadata":{},"id":"034b3075-68bf-4bfe-85aa-b6d5c35ffba8"},{"cell_type":"markdown","source":"2.19 How many ASVs have more than 1,000 reads?","metadata":{},"id":"5ccfb0c5-77e3-4907-883d-b724588ad9b2"},{"cell_type":"markdown","source":"7","metadata":{},"id":"41b16781-0efb-45a2-ab0c-aaf8014533e5"},{"cell_type":"markdown","source":"2.20 In how many samples is present the most frequent ASV? ","metadata":{},"id":"af43270c-4b38-45e2-b019-64940b7ffed8"},{"cell_type":"markdown","source":"39","metadata":{},"id":"029b4534-8bd5-4797-af81-19b6096d39c7"},{"cell_type":"markdown","source":"2.21 What name has this ASV?","metadata":{},"id":"62ac4145-c149-492e-8928-1ec6c8eb4a59"},{"cell_type":"markdown","source":"445008ecee377be1d073769dbc4e6d0a","metadata":{},"id":"cb35e851-d476-4cd7-979e-e32f298f30c7"},{"cell_type":"markdown","source":"2.22 Convert the `representative-sequences.qza` file to `qzv`.","metadata":{},"id":"d3fee216-97ac-4961-aeba-26424702f228"},{"cell_type":"code","source":"qiime feature-table tabulate-seqs \\\n  --i-data          rep-seqs.qza \\\n  --o-visualization rep-seqs.qzv","metadata":{"tags":[],"trusted":true},"execution_count":7,"outputs":[{"name":"stdout","text":"\u001b[32mSaved Visualization to: rep-seqs.qzv\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"cd8ec6eb-c771-46bb-a9f9-651d0031d32b"},{"cell_type":"markdown","source":"2.23 How long is the longest ASV?","metadata":{},"id":"3c07785d-505b-472d-a56b-e773092dc3f3"},{"cell_type":"markdown","source":"The longest ASV is 210 nucleotides long","metadata":{},"id":"b2b11216-0c9c-4611-9706-8193056424d0"},{"cell_type":"markdown","source":"2.24 And the shortest?","metadata":{},"id":"81e15343-b84e-4e9e-9370-b649739f7ad4"},{"cell_type":"markdown","source":"The shortest ASV is 153 nucleotides long","metadata":{},"id":"948883c4-a459-42da-a519-b4adf5757592"},{"cell_type":"markdown","source":"2.25 How long is the most common ASV?","metadata":{},"id":"33a18e39-ef69-466d-8e2c-e2ca7fb8d880"},{"cell_type":"markdown","source":"174 nucleotides long","metadata":{},"id":"17e796c5-41c9-4327-ac72-78eb4ea0e578"},{"cell_type":"markdown","source":"2.26 Click on the sequence and perform a BLAST search. What is the first result?\n\n**Note**: this step will take a while. Come back later to finish this.","metadata":{"tags":[]},"id":"f26ec314-9791-493c-93c3-65b2672dd059"},{"cell_type":"markdown","source":"Junceella aquamata 18S ribosomal RNA gene, partial sequence, AY962535.1","metadata":{},"id":"aa4430ac-3555-44ef-94f9-07c71176ecbd"},{"cell_type":"markdown","source":"2.27 Visualize the `dada2-stats` file.","metadata":{},"id":"c1488733-8488-4960-b6b5-9dfb21339310"},{"cell_type":"code","source":"qiime metadata tabulate \\\n    --m-input-file    dada2-stats.qza \\\n    --o-visualization dada2-stats.qzv","metadata":{"tags":[],"trusted":true},"execution_count":8,"outputs":[{"name":"stdout","text":"\u001b[32mSaved Visualization to: dada2-stats.qzv\u001b[0m\n\u001b[0m\n","output_type":"stream"}],"id":"4873e001-2cf6-4f74-a821-ba9bfa45d1b5"},{"cell_type":"markdown","source":"2.28 Which sample has been filtered the most in terms of percentages? Type the sample name and the percentage.","metadata":{},"id":"63e7c74f-11a0-4792-9f37-eeb0869070fb"},{"cell_type":"markdown","source":"sardine_capbreton_outer_day_XX11 (77.44)","metadata":{},"id":"34604990-53ce-422e-b61d-0095cf44831d"},{"cell_type":"markdown","source":"2.29 And the least?","metadata":{},"id":"26e836a2-5018-4485-bbe1-6df53d12eea8"},{"cell_type":"markdown","source":"sardine_gironde_inner_night_L14 (3.82)","metadata":{},"id":"33f180c1-269c-4fa9-8250-02eb67e5de09"},{"cell_type":"markdown","source":"It should be a great idea to save your notebook to your laptop, press the Fast Forward button to see that all files generate properly, `pause` the notebook, and take 5 minutes away from the screen.","metadata":{},"id":"b5bc2b7b-82ad-4e0e-9a69-5d31a719e8a7"},{"cell_type":"code","source":"pause","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"2d45e117-8e35-4c88-bd81-6cbfe3bef264"},{"cell_type":"markdown","source":"## 3. Taxonomy","metadata":{},"id":"b60484f5-424e-43b6-a770-1bd769de0f5a"},{"cell_type":"markdown","source":"In the folder databases you have two trained sequence classifiers:\n- greengenes, which contains 16S sequences \n- metazoogene, which contain 18S sequences","metadata":{},"id":"85540b16-d5fa-4f20-9914-0cd4415e72d1"},{"cell_type":"code","source":"ll databases/","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"73fd46c9-f61f-4761-a6c1-fd451385db7e"},{"cell_type":"markdown","source":"3.1 Which one is appropriate to analyze the eukaryotes present in the guts of these fish?","metadata":{},"id":"7a16bd85-897a-4ece-966e-d99581dddf97"},{"cell_type":"markdown","source":"18S ","metadata":{},"id":"9896607c-7350-4640-9745-a23e30ebf942"},{"cell_type":"markdown","source":"3.2 Classify your sequences with that database.","metadata":{},"id":"6cb5b109-30fb-4520-b219-a4ce80c8e1fa"},{"cell_type":"code","source":"qiime feature-classifier classify-sklearn \\\n    --i-classifier     databases/metazoogene.qza \\\n    --i-reads          rep-seqs.qza \\\n    --o-classification taxonomy.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"b14722fc-3ea0-4a17-80c5-6451608c63fa"},{"cell_type":"markdown","source":"3.3 Convert the results into a visualization","metadata":{},"id":"c9301078-2f8a-4fec-94b1-ef02b3ea15b9"},{"cell_type":"code","source":"qiime metadata tabulate \\\n    --m-input-file    taxonomy.qza \\\n    --o-visualization taxonomy.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"eeb369e3-c6ec-4d74-b89c-0d327ab70899"},{"cell_type":"markdown","source":"3.4 How many ASVs have been classified as Animalia and no phylum?","metadata":{},"id":"f29ccd18-3ad3-48cf-9101-3fe0dafd50c1"},{"cell_type":"markdown","source":"35","metadata":{},"id":"fa98fd30-b3dc-4093-b7a7-4772ceadbf0e"},{"cell_type":"markdown","source":"3.5 How many belong to the class Copepoda?","metadata":{},"id":"9aa77994-e3d4-4e57-b848-05ad1e53e088"},{"cell_type":"markdown","source":"16","metadata":{},"id":"fb0e16a8-fd98-41e4-a301-4dc5edcd7c4e"},{"cell_type":"markdown","source":"3.6 Draw the taxonomic barplot.","metadata":{},"id":"bef456cf-c420-48d9-8a60-ee5d82e5572e"},{"cell_type":"code","source":"qiime taxa barplot \\\n  --i-table         table.qza \\\n  --i-taxonomy      taxonomy.qza \\\n  --m-metadata-file guts_night_outer.tsv \\\n  --o-visualization taxa-bar-plots.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"80f4d7e6-0d16-44c2-8edc-d122f6a15912"},{"cell_type":"markdown","source":"3.7 What sample contains the most of the Engraulis encrasicolus species?","metadata":{},"id":"4ccd21ce-ce03-4c0c-908c-6cfb973a1424"},{"cell_type":"markdown","source":"sardine_capbreton_outer_day_L31.10 (3.130%)","metadata":{},"id":"1db33663-0e61-4058-be36-e2d6f644ef45"},{"cell_type":"markdown","source":"3.8 What is the most common phylum (in average)?","metadata":{},"id":"401f1683-be08-4d2c-9225-601a5e61f4d4"},{"cell_type":"markdown","source":"Arthropoda","metadata":{},"id":"d8d7f62f-1422-4d36-8c8e-6e007d8c99fa"},{"cell_type":"markdown","source":"## 4. Diversity","metadata":{},"id":"0395eb9c-0e58-4604-a619-8dc5055134e3"},{"cell_type":"markdown","source":"4.1 Create first the phylogeny, and then compute the core metrics. Filter out samples with less than 1,000 reads.","metadata":{},"id":"5a7506c5-44d9-4019-9199-e64442d12341"},{"cell_type":"code","source":"qiime phylogeny align-to-tree-mafft-fasttree \\\n    --i-sequences        rep-seqs.qza \\\n    --o-alignment        phylo-aligned-seqs.qza \\\n    --o-masked-alignment phylo-masked-aligned-seqs.qza \\\n    --o-tree             phylo-unrooted-tree.qza \\\n    --o-rooted-tree      phylo-rooted-tree.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"ed641867-e419-4ab3-9d3c-2c1896dd3b57"},{"cell_type":"code","source":"export UNIFRAC_USE_GPU=N  # There is an error if you have a GPU\nqiime diversity core-metrics-phylogenetic \\\n    --i-phylogeny      phylo-rooted-tree.qza \\\n    --i-table          table.qza \\\n    --p-sampling-depth 1000 \\\n    --m-metadata-file  guts_night_outer.tsv \\\n    --output-dir       metrics","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"1035e21a-41ae-411d-9a2b-7f40e0defbae"},{"cell_type":"markdown","source":"4.2 Compute the alpha-rarefaction, using a maximum depth of 2000 sequences.","metadata":{},"id":"dbcd1546-4447-435b-9f8a-2abfd0337355"},{"cell_type":"code","source":"\nqiime diversity alpha-rarefaction \\\n    --i-table         table.qza \\\n    --i-phylogeny     phylo-rooted-tree.qza \\\n    --p-max-depth     2000 \\\n    --m-metadata-file guts_night_outer.tsv \\\n    --o-visualization metrics/alpha-rarefaction.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"a55c3433-c829-4a15-9481-0f90ee61538b"},{"cell_type":"markdown","source":"4.3 According to the Shannon index, do we have almost the same community information at a sequencing depth of 1,000 and 2,000?","metadata":{},"id":"c9253ffa-802c-4eec-b2d8-f3a2e6d79b6a"},{"cell_type":"markdown","source":"There is a similar Shannon diversity for most of the barcodes, although some are not represented at higher sequence depths.","metadata":{},"id":"261f0e00-0f15-411b-8b0a-b12032ef3208"},{"cell_type":"markdown","source":"4.4 Which barcode has the least Shannon diversity?","metadata":{},"id":"77e020f4-bc26-428e-be4d-f207792b71f3"},{"cell_type":"markdown","source":"CTCCCTTTGTGT","metadata":{},"id":"84a8ea86-4018-4079-8385-e689c0867f03"},{"cell_type":"markdown","source":"4.5 Compute the evenness of the samples.","metadata":{},"id":"a00d2f86-adae-4c78-98e2-b6d98f1bcbdf"},{"cell_type":"code","source":"# evenness\nqiime diversity alpha-group-significance \\\n  --i-alpha-diversity metrics/evenness_vector.qza \\\n  --m-metadata-file   guts_night_outer.tsv \\\n  --o-visualization   metrics/evenness-group-significance.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"e67c9554-b2d4-4f5f-9922-0a217cdb121f"},{"cell_type":"markdown","source":"4.6 Do the same for the richness.","metadata":{},"id":"96c6527c-90b8-4408-b598-66426b47ba33"},{"cell_type":"code","source":"# Richness ~ faith phylogenetic diversity\nqiime diversity alpha-group-significance \\\n  --i-alpha-diversity metrics/faith_pd_vector.qza \\\n  --m-metadata-file   guts_night_outer.tsv \\\n  --o-visualization   metrics/faith-pd-group-significance.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"9a178c20-39cd-4f16-8d74-9aefcaf8581c"},{"cell_type":"markdown","source":"4.7 Is the richness different between your types of fish? What is your p-value?","metadata":{},"id":"84a49915-2eb2-4bf0-b3bc-20e2faae9d22"},{"cell_type":"markdown","source":"The richness is different since H: 7.806, p:0.005 > (0.05)","metadata":{},"id":"82546803-79f1-43df-93f2-1782a1384fd1"},{"cell_type":"markdown","source":"4.8 Is the evenness different between fish type? What is your p-value?","metadata":{},"id":"181daaaa-cb65-406b-b041-a447a7c6a0e9"},{"cell_type":"markdown","source":"The evenness is different since H: 6.668, p: 0.001 > (0.05)","metadata":{},"id":"82af6084-a850-464f-9367-f04a01f7de17"},{"cell_type":"markdown","source":"4.9 Compute the beta diversity using the Weighted Unifrac distance matrix, grouping by fish type.","metadata":{},"id":"91619e50-ed56-4ac1-8d73-82fa06a2127a"},{"cell_type":"code","source":"\nqiime diversity beta-group-significance \\\n    --i-distance-matrix metrics/weighted_unifrac_distance_matrix.qza \\\n    --m-metadata-file   guts_night_outer.tsv \\\n    --m-metadata-column sample_type \\\n    --o-visualization   metrics/weighted-unifrac-body-site-significance.qzv \\\n    --p-pairwise\n     ","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"2449ee10-8bf5-47b4-ad34-b5fb15f40dab"},{"cell_type":"markdown","source":"4.10 According to the pairwise Permanova test, are there differences between fish types?","metadata":{},"id":"d3774cb8-2db6-4216-be1b-9b0c5e37b470"},{"cell_type":"markdown","source":"Yes there are differences, test statistic: 9.05, p-value: 0.001> (0.05)","metadata":{},"id":"d4a29637-5256-45ef-bbcd-fbd62b73ba87"},{"cell_type":"markdown","source":"`pause` and take 5 minutes if you need it. Maybe it is time to collect that BLAST result from before.","metadata":{},"id":"b845380b-f34b-474e-b39e-cb4a5fc5236b"},{"cell_type":"code","source":"pause","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"76b44b7b-c662-440f-9f55-4b4383f35a3b"},{"cell_type":"markdown","source":"## 5 Analysis of Composition with ANCOM","metadata":{},"id":"b18a2d7c-b6e7-47d4-9d0b-a3fec050ff9d"},{"cell_type":"markdown","source":"5.1 Do the analysis of composition at the ASV level, and compare the differences of composition between **seasons**. You have to perform the `add-pseudocount` step first.","metadata":{},"id":"7adcbc98-9ae1-4e0f-93f3-d1af4772bb85"},{"cell_type":"code","source":"qiime feature-table filter-samples \\\n    --i-table          table.qza \\\n    --m-metadata-file  guts_night_outer.tsv \\\n    --o-filtered-table region.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"024faa1e-8ba6-4d94-8b0e-8c6a95faefc8"},{"cell_type":"code","source":"\nqiime composition add-pseudocount \\\n  --i-table             region.qza \\\n  --o-composition-table comp-region.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"7d368344-1398-4581-b40c-aa94cbbdae59"},{"cell_type":"code","source":"qiime composition ancom \\\n    --i-table           comp-region.qza \\\n    --m-metadata-file   guts_night_outer.tsv \\\n    --m-metadata-column region \\\n    --o-visualization   ancom-region.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"3e0a2642-6799-45db-9c26-64a955c76446"},{"cell_type":"markdown","source":"5.2 How many ASVs are statistically different? ","metadata":{},"id":"17fb6d31-7e54-4081-92d3-c4b5e991abb5"},{"cell_type":"markdown","source":"11","metadata":{},"id":"6589416a-672e-41f0-b95a-8f0654e9be6b"},{"cell_type":"markdown","source":"5.3 How many have a positive CLR (centered log ratio)?","metadata":{},"id":"12b00319-3504-42d0-8816-0bb5ef82fd15"},{"cell_type":"markdown","source":"6","metadata":{},"id":"84b8921c-684d-46b6-b4b3-ec46679c5578"},{"cell_type":"markdown","source":"5.4 How many are negative?","metadata":{},"id":"878c9d92-ce2d-449a-8305-c8ab1307ecf1"},{"cell_type":"markdown","source":"5","metadata":{},"id":"21ce8f8e-43c0-43ec-835b-d369a658b537"},{"cell_type":"markdown","source":"5.5 Colapse the ASV table so we can compute the results at the family level","metadata":{},"id":"d3a5db98-d69f-4ec6-b263-116765b7f8c6"},{"cell_type":"code","source":"qiime taxa collapse \\\n  --i-table           region.qza \\\n  --i-taxonomy        taxonomy.qza \\\n  --p-level           5 \\\n  --o-collapsed-table gut-table-5.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"1c5a1034-a6e5-4d45-800e-5869d0869524"},{"cell_type":"markdown","source":"5.6  Compute the pseudocounts for that table, and run the ANCOM analysis again between seasons.","metadata":{},"id":"dcae29fd-12f4-45ec-b742-29e0432a5fb1"},{"cell_type":"code","source":"qiime composition add-pseudocount \\\n  --i-table             gut-table-5.qza \\\n  --o-composition-table comp-gut-table-5.qza","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"819e6b88-faa9-49ae-9aae-ff90afab2cca"},{"cell_type":"code","source":"qiime composition ancom \\\n  --i-table           comp-gut-table-5.qza \\\n  --m-metadata-file   guts_night_outer.tsv \\\n  --m-metadata-column region \\\n  --o-visualization   region-ancom-subject.qzv","metadata":{"tags":[],"trusted":true},"execution_count":null,"outputs":[],"id":"61518013-7cf5-4764-b52a-52510d94d42c"},{"cell_type":"markdown","source":"5.7  How many elements are statistically different now?","metadata":{},"id":"73798f3a-962a-49bb-aca9-5b411b90f5f0"},{"cell_type":"markdown","source":"7","metadata":{},"id":"92ac0927-ba7e-4b4f-b6c3-0afa66d749b6"},{"cell_type":"markdown","source":"## End of the notebook","metadata":{"tags":[]},"id":"62484b62-aae7-4733-8bd1-34ef80b68713"},{"cell_type":"markdown","source":"Make sure that\n- all the text cells contain only text\n- all the code cells contain only code\n- That the entire notebook runs from the beginning to the end without errors by pressing the Fast Forward button.","metadata":{},"id":"a68435ea-682f-4869-8ab4-523a883bdca6"}]}